src/mixins/ContentMixin.js

```diff
@@ -60,13 +61,33 @@ export default {
       };
 
       if (type == "MultiFlatmap" || type == "Flatmap") {
+        console.log(resource)
         result.internalName = this.idNamePair[resource.feature.models];
         if (resource.eventType == "click") {
           result.eventType = "selected";
           if (resource.feature.type == "marker") {
-            returnedAction = {};
-            returnedAction.type = "Facet";
-            returnedAction.label = this.idNamePair[resource.feature.models];
+            let label = this.idNamePair[resource.feature.models]
+            window.label = label
+            window.id = resource.feature.models
+            let hardcodedAnnotation = markerZoomLevels.filter(mz=>mz.id === resource.feature.models)
+            window.ha = hardcodedAnnotation
+            if( hardcodedAnnotation.filter(h=>h.keyword).length > 0){
+              // Keyword searches do not contain labels, so switch to keyword search if no label exists
+              returnedAction = {
+                type: "Search",
+                term: 'http://purl.obolibrary.org/obo/' + resource.feature.models.replace(':','_')
+              };
+              window.ra = returnedAction
+            } else {
+            // Facet search on anatomy if it is not a keyword search
+              returnedAction = {
+                type: "Facet",
+                facet: label,
```

Comment:
label is undefined here, I used to call this.idNamePair[resource.feature.models] to get the label instead.